### PR TITLE
Add interface for shared library

### PR DESCRIPTION
**Pull request by Flanker32**
Add interface for shared library

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServiceCreator.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import java.util.Map;
+
+public interface IAppServiceCreator<T> {
+    IAppServiceCreator<T> withName(String name);
+
+    IAppServiceCreator<T> withSubscription(String subscriptionId);
+
+    IAppServiceCreator<T> withResourceGroup(String resourceGroupName);
+
+    IAppServiceCreator<T> withAppServicePlan(String appServicePlanId);
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServiceCreator.java:22]: withPlan?

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServiceCreator.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+    IAppServiceCreator<T> withAppServicePlan(String resourceGroup, String planName);
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServiceCreator.java:24]: no need for this one

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServicePlan.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+import com.microsoft.azure.toolkits.appservice.entity.AppServicePlanEntity;
+
+public interface IAppServicePlan {
+    IAppServicePlanCreator create();
+
+    IAppServicePlanUpdater update();
+
+    boolean exists();
+
+    AppServicePlanEntity get();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServicePlan.java:17]: entity

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServicePlan.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+    com.azure.resourcemanager.appservice.models.AppServicePlan getAppServicePlanClient();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServicePlan.java:19]: delete it

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServicePlanUpdater.java [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+ */
+package com.microsoft.azure.toolkits.appservice.service;
+
+import com.microsoft.azure.toolkits.appservice.model.PricingTier;
+import com.microsoft.azure.toolkits.appservice.service.serviceplan.AppServicePlan;
+
+public interface IAppServicePlanUpdater {
+    IAppServicePlanUpdater withPricingTier(PricingTier pricingTier);
+
+    AppServicePlan apply();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IAppServicePlanUpdater.java:14]: commit

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebApp.java [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+    void delete();
+
+    void deploy(File file);
+
+    void deploy(DeployType deployType, File file);
+
+    boolean exists();
+
+    WebAppEntity get();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebApp.java:32]: entity

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebApp.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+
+    WebApp.WebAppUpdater update();
+
+    Runtime getRuntime();
+
+    PublishingProfile getPublishingProfile();
+
+    IWebAppDeploymentSlot deploymentSlot(String slotName);
+
+    List<IWebAppDeploymentSlot> deploymentSlots();
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebApp.java:44]: these method should be invoked by the client with WebAppEnitity.getRuntime()/getPublishingProfile()/... ?
how about remove them in this interface.

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebAppDeploymentSlot.java [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+    void restart();
+
+    void delete();
+
+    void deploy(File file);
+
+    void deploy(DeployType deployType, File file);
+
+    boolean exists();
```
Comment (shenqianjin) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebAppDeploymentSlot.java:28]: what's the meaning of "exists()" ? Is it "equals to "get() = null" ?

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebApp.java (hunk 7 above)
Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebApp.java:44]: Yes, but some properties which is widely used in our tools, so may be we need to move some common parameters to interface.

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebAppDeploymentSlot.java (hunk 8 above)
Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebAppDeploymentSlot.java:28]: get() will return the entity directly, but exists() will check whether the resource exists in Azure

File: azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebAppDeploymentSlotCreator.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+
+public interface IWebAppDeploymentSlotCreator {
+
+    IWebAppDeploymentSlotCreator withName(String name);
+
+    IWebAppDeploymentSlotCreator withAppSettings(Map<String, String> appSettings);
+
+    IWebAppDeploymentSlotCreator withConfigurationSource(String source);
+
+    IWebAppDeploymentSlot create();
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkits/appservice/service/IWebAppDeploymentSlotCreator.java:19]: commit

**Review by andxu — APPROVED** [review]

**Pull request merged**